invertedai/large/initialize.py

```diff
@@ -0,0 +1,391 @@
+
+        area_size = tuple([width,height])
+
+    regions = iai.get_regions_in_grid(
+        width = area_size[0], 
+        height = area_size[1],
+        map_center = map_center
+    )
+
+    if total_num_agents is None:
+        total_num_agents = 10*len(regions)
+
+    new_regions = iai.get_number_of_agents_per_region_by_drivable_area(
+        location = location,
+        regions = regions,
+        total_num_agents = total_num_agents,
+        random_seed = random_seed,
+        display_progress_bar = display_progress_bar
+    )
+
+    return new_regions
+
+@validate_call
+def get_regions_in_grid(
+    width: float,
+    height: float,
+    map_center: Optional[Tuple[float,float]] = (0.0,0.0), 
+    stride: Optional[float] = 50.0
+) -> List[Region]:
+    """
+    A utility function to help initialize an area larger than 100x100m. This function breaks up an 
+    area into a grid of 100x100m regions, each of which can be initialized directly.
+
+    Arguments
+    ----------
+    width:
+        The horizontal size of the area which will be broken into smaller regions.
+
+    height:
+        The vertical size of the area which will be broken into smaller regions.
+
+    map_center:
+        The coordinates of the center of the rectangular area to be broken into smaller regions.
+
+    stride:
+        How far apart the centers of the 100x100m regions should be. Some overlap is recommended for best results.
+    
+    """
+    def check_valid_center(center):
+        return (map_center[0] - width) < center[0] < (map_center[0] + width) and \
+            (map_center[1] - height) < center[1] < (map_center[1] + height)
+
+    def get_neighbors(center):
+        return [
+            (center[0] + (i * stride), center[1] + (j * stride))
+            for i, j in list(product(*[(-1, 1),]* 2))
+        ]
+
+    queue, centers = [map_center], []
+
+    while queue:
+        center = queue.pop(0)
+        neighbors = filter(check_valid_center, get_neighbors(center))
+        queue.extend([
+            neighbor
+            for neighbor in neighbors
+            if neighbor not in queue and neighbor not in centers
+        ])
+        if center not in centers and check_valid_center(center):
+            centers.append(center)
+    
+    regions = [None for _ in range(len(centers))]
+    for i, center in enumerate(centers):
+        regions[i] = Region.init_square_region(center=Point.fromlist(list(center)))
+
+    return regions
+
+@validate_call
+def get_number_of_agents_per_region_by_drivable_area(
+    location: str,
+    regions: List[Region],
+    total_num_agents: Optional[int] = 10,
+    random_seed: Optional[int] = None,
+    display_progress_bar: Optional[bool] = True
+) -> List[Region]:
+    """
+    This function takes in a list of regions, infers the total drivable surface area per region by calculating 
+    the number of black pixels (non-drivable surfaces), then inserts a number of default AgentAttributes objects
```

Comment:
```suggestion
    This function takes in a list of regions, calculates the driveable area for each of them
    using :func:`location_info`, then inserts a number of default AgentAttributes objects
```
The implementation details are just confusing, but we want to let the user know which API endpoints will be called.

Reply:
Done.